### ISSUE TITLE
Delaying runTiming node flickering

### ISSUE THREAD

**Issue by eleddie** [wcandillon/react-native-redash#89]
I need to run some timing parallel one to the other, I managed to do this pretty easy with multiple `runTiming`s. Now I need to add delay to some of them. My logic told me to use `runDelay` with a `runTiming` inside it but im getting a flicker effect.

This is one of the animations without the delay (note that theres an opacity an translate animation):
```javascript
const cardSetup = animSetup(800, 1);
this.opacityCard = runTiming(cardSetup.clock, 0, cardSetup.config);
this.translateCard = bInterpolate(this.opacityCard, 0, -150);
```

The `animSetup` function is a helper function which contains this:
```javascript
const animSetup = (duration, toValue) => {
  return {
    clock: new Clock(),
    config: {
      duration,
      toValue,
      easing: Easing.bezier(0.25, 0.1, 0.25, 1),
    },
  };
}
```

And it looks like this (works perfectly fine):
![runTiming](https://user-images.githubusercontent.com/13477629/62737095-a309d480-b9f4-11e9-8045-7fb44da8968d.gif)

Now if I add the `runDelay` function:
```javascript
const cardSetup = animSetup(800, 1);
this.opacityCard = runDelay(runTiming(cardSetup.clock, 0, cardSetup.config), 100);
this.translateCard = bInterpolate(this.opacityCard, 0, -150);
```

It looks like this:
![runDelayBug](https://user-images.githubusercontent.com/13477629/62737275-0431a800-b9f5-11e9-8c13-8a5ed2f1ff96.gif)

I'm not sure if it is a bug on the `runDelay` function, a bug on `react-native-reanimated` or maybe a wrong way to do it.

Any idea how to accomplish this effect?

**Comment by rodolfovilaca** [comment]
Idk how to explain the real problem itself, if the delay which is another `runTiming` is delaying every set of the node but doesn't stop the animation to run, so the effect is the first position of the node and then sets the position after the delay.

In other words i think what you need is to delay the `startClock` of the timing animation, for this you would have to change a little bit the runTiming function to accept a delay duration like the example below:

```javascript
export function runTiming(clock, value, config, delay) {
  const state = {
    finished: new Value(0),
    position: new Value(0),
    time: new Value(0),
    frameTime: new Value(0)
  };

  return block([
    onChange(config.toValue, set(state.frameTime, 0)),
    cond(clockRunning(clock), 0, [
      set(state.finished, 0),
      set(state.time, 0),
      set(state.position, value),
      set(state.frameTime, 0),
      runDelay(startClock(clock), delay)
    ]),
    timing(clock, state, config),
    cond(state.finished, stopClock(clock)),
    state.position
  ]);
}
```

From this you should be able to call:

```javascript
const cardSetup = animSetup(800, 1);
this.opacityCard = runTiming(cardSetup.clock, 0, cardSetup.config, 100)
this.translateCard = bInterpolate(this.opacityCard, 0, -150);
```

Maybe we could make a proposal so that `timing`, `spring`, and `decay` animations should accept a delay prop to delay the `startClock` of each animation.

**Comment by wcandillon** [comment]
there is a delay function available, did you try it?

**Comment by rodolfovilaca** [comment]
Yes i believe he was using a version previous to `8.0.0` where the runners were renamed, and he called the `runDelay` runner.

You can see from this [expo-snack](https://snack.expo.io/@rodolfovilaca/delay-example) that i built to reproduce that effect what he is talking about and my solution to delaying the clock to start.

**Comment by luogao** [comment]
The delay function may need a clock argument, in case combine with other function, like loop function

I have tried run an animation use delay & loop and I made it works like below:
``` javascript
animationValue = block([
    timing({ clock: this.clock, from: 0, to: 1, duration: this.props.delay || 0 }),
    cond(not(clockRunning(this.clock)), [ loop({ clock: this.clock, duration: 6000, easing: Easing.linear })])
])


```

and the source code of delay use an internal `new Clock` which may case problem

```javascript
export const delay = (node: Animated.Node<number>, duration: number) => {
  const clock = new Clock();
  return block([
    timing({ clock, from: 0, to: 1, duration }),
    cond(not(clockRunning(clock)), node)
  ]);
};


```

**Comment by luogao** [comment]
> The delay function may need a clock argument, in case combine with other function, like loop function
> 
> I have tried run an animation use delay & loop and I made it works like below:
> 
> ```js
> animationValue = block([
>     timing({ clock: this.clock, from: 0, to: 1, duration: this.props.delay || 0 }),
>     cond(not(clockRunning(this.clock)), [ loop({ clock: this.clock, duration: 6000, easing: Easing.linear })])
> ])
> ```
> 
> and the source code of delay use a `new Clock` in itself which may cause problem
> 
> ```js
> export const delay = (node: Animated.Node<number>, duration: number) => {
>   const clock = new Clock();
>   return block([
>     timing({ clock, from: 0, to: 1, duration }),
>     cond(not(clockRunning(clock)), node)
>   ]);
> };
> ```

Seems like this solution may have performance issues, 😥😥😥

**Comment by Thram** [comment]
@eleddie 

I had the same problem and created this 2 functions that solved the problem:
https://gist.github.com/Thram/e9ca9de3e4b47b9877f5bd089b5059ea
You can use them like:
```js
const animation = new Value(0);
useCode(() => set(animation, delay(loop({ duration }), delayTime)), [
    animation,
    duration,
    delayTime,
  ]);
```
